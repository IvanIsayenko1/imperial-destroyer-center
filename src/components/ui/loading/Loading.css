.loading-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 2rem;
  gap: 1rem;
  height: 100%;
}

.loading-spinner {
  width: 40px;
  height: 40px;
  border: 3px solid rgba(255, 30, 34, 0.1);
  border-radius: 50%;
  border-top-color: var(--accent-color);
  animation: spin 1s ease-in-out infinite;
}

.loading-text {
  color: white;
  font-size: 0.875rem;
  margin: 0;
}

@keyframes spin {
  to {
    transform: rotate(360deg);
  }
}
