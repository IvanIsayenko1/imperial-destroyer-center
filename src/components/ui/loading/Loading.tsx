import React from "react";
import "./Loading.css";

interface LoadingProps {
  text?: string;
}

export const Loading: React.FC<LoadingProps> = ({ text = "Loading..." }) => {
  return (
    <div className="loading-container">
      <div className="loading-spinner" />
      <p className="loading-text">{text}</p>
    </div>
  );
};
