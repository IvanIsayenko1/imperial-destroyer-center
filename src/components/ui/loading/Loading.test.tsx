import { render, screen } from "@testing-library/react";
import { Loading } from "./Loading";

describe("Loading component", () => {
  it("should render the loading spinner", () => {
    render(<Loading />);
    const spinner = screen.getByTestId("loading-spinner");
    expect(spinner).toBeInTheDocument();
  });

  it("should render the default loading text when no text prop is provided", () => {
    render(<Loading />);
    const text = screen.getByText("Loading...");
    expect(text).toBeInTheDocument();
  });

  it("should render custom text when the text prop is provided", () => {
    const customText = "Please wait...";
    render(<Loading text={customText} />);
    const text = screen.getByText(customText);
    expect(text).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(<Loading />);
    expect(asFragment()).toMatchSnapshot();
  });
});
