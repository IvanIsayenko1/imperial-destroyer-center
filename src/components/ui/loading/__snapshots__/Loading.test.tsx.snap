// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loading component > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="loading-container"
  >
    <div
      class="loading-spinner"
      data-testid="loading-spinner"
    />
    <p
      class="loading-text"
    >
      Loading...
    </p>
  </div>
</DocumentFragment>
`;
